transports/quic/src/config.rs

```diff
@@ -59,6 +59,8 @@ pub struct Config {
     /// If support for draft-29 is enabled servers support draft-29 and version 1 on all
     /// QUIC listening addresses.
     /// As client the version is chosen based on the remote's address.
+    #[allow(deprecated)]
```

Comment:
I think the `#[allow(deprecated)]` needs to be applied in all the places where `support_draft_29` is used, instead of here where it is applied.

Also, could we use `expect` instead of `allow`? `expect` will give us a warning as soon as the attribute isn't needed anymore, i.e. as soon as the deprecated items are removed.
(Depends on #5650 to update our MSRV to the most recent rust version where `expect` was stablilized.)

Reply:
```suggestion
```